types/d3/v3/index.d.ts

```diff
@@ -2605,7 +2609,8 @@ declare namespace d3 {
             y(): brush.Scale;
             y(y: brush.Scale): Brush<T>;
 
-            extent(): [number, number] | [[number, number], [number, number]];
+            // https://github.com/d3/d3-3.x-api-reference/blob/master/SVG-Controls.md#brush_extent
+            extent(): [number, number] | [[number, number], [number, number]] | [Date, Date];
```

Comment:
I have a couple of issues with this:

1. The return type isn't consistent -- there's no handling of the type `[[Date, Date], [Date, Date]]`.

2. The `extent` parameter in the overload below this doesn't accept the new return type, so I can't pass the extent through a round trip:

    ```ts
    brush.extent(brush.extent()); // [Date, Date] is not compatible with [number, number]
    ```

3. The new Date-returning case (which I suppose happens if you use `d3.time.scale()`) doesn't appear to be actually possible, because the `brush.Scale` interface is defined to [only work on numeric domains and ranges](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/d3/v3/index.d.ts#L2583-L2593). In order to use a time scale here, that interface would also need changing (and I'm not sure what the best idea would be here, using a union of two interfaces or a single interface that worked on the type `number | Date`).

Reply:
This change of mine was implemented indeed short sighted. The necessary changes include an extension to `Scale` as well as the `extent` parameters. 

I did those changes and can commit them for you to check. It is necessary to change the definitions because it is otherwise prohibitive for Date Axis.